### PR TITLE
Knn

### PR DESCRIPTION
**Pull request by mayurdb**
Added naive implementation of KNN

### CODE REVIEW

File: src/main/scala/com/spark3d/spatialOperator/SpatialQuery.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    val itr = rdd.toLocalIterator
+
+    while (itr.hasNext) {
+      val currentElement = itr.next
+      if (pq.size < k) {
+        pq.enqueue(currentElement)
+      } else {
+        val currentEleDist = currentElement.center.distanceTo(queryObject.center)
+        // TODO make use of pq.max
```
Comment (mayurdb) [inline src/main/scala/com/spark3d/spatialOperator/SpatialQuery.scala]: on using pq.max, getting no implicit ordering defined for class B, even though we have defined one in GeometryObjectComparator. Any idea how we can resolve this???

Comment (JulienPeloton) [reply src/main/scala/com/spark3d/spatialOperator/SpatialQuery.scala]: Hum, good question. Let me have a look at this today.

**Comment by JulienPeloton** [comment]
Can I merge those changes, or you want to push more?

**Comment by mayurdb** [comment]
Yes, we can merge this. I will work on the efficient implementation separately, would be easier to revert in case of any issue ;p

**Comment by JulienPeloton** [comment]
Alright!

File: src/main/scala/com/spark3d/spatialOperator/SpatialQuery.scala (hunk 1 above)
Comment (mayurdb) [reply src/main/scala/com/spark3d/spatialOperator/SpatialQuery.scala]: I think I tried this, let me give this another shot. Will give an update here

**Pull request merged**